packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -184,11 +181,11 @@ class RequestWrapper {
       assert.isInstance({request}, Request);
     }
 
-    const clonedRequest = this.pluginCallbacks.fetchDidFail ?
+    const clonedRequest = this.plugins.has('fetchDidFail') ?
```

Comment:
Nit: Can you move this to where it's used.

Reading this, you check for fetchDidFail, clone the request, then iterate over requestWillFetch (which threw me because I assumed it had failed), hence the cloning.